diff --git BUILD.gn BUILD.gn
index 1edcbcee..03b4c3a6 100644
--- BUILD.gn
+++ BUILD.gn
@@ -20,7 +20,7 @@ group("nwjs") {
     deps += [ "//buildtools/third_party/libc++:libcpp" ]
   }
   if (is_linux) {
-    deps += [ "//chrome:xdg_mime", "//components/crash/core/app:chrome_crashpad_handler", ":copy_icd" ]
+    deps += [ "//chrome:xdg_mime", "//components/crash/core/app:chrome_crashpad_handler" ]
   }
 }
 
@@ -260,7 +260,7 @@ action("commit_id") {
     public_configs = [ ":commit_id_config" ]
 }
 
-if (is_linux) {
+if (is_linux && target_cpu != "loong64") {
   copy("copy_icd") {
     sources = [
       "$root_out_dir/vk_swiftshader_icd.json",
@@ -386,6 +386,16 @@ if (is_linux && !is_component_build) {
              "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL",
              "//third_party/swiftshader/src/OpenGL/libGLESv2:swiftshader_libGLESv2"
              ]
+    if (target_cpu == "loong64") {
+      sources -= [
+        "$root_build_dir/swiftshader/libEGL.so",
+        "$root_build_dir/swiftshader/libGLESv2.so"
+      ]
+      deps -= [
+        "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL",
+        "//third_party/swiftshader/src/OpenGL/libGLESv2:swiftshader_libGLESv2"
+      ]
+    }
     if (target_cpu == "x64") {
       inputs = [
         "$root_build_dir/dump_syms",
diff --git tools/package_binaries.py tools/package_binaries.py
index bc42b326..bd5aea08 100755
--- tools/package_binaries.py
+++ tools/package_binaries.py
@@ -155,24 +155,12 @@ def generate_target_nw(platform_name, arch, version):
                            'lib/libEGL.so',
                            'lib/libGLESv2.so',
                            'lib/libvulkan.so.1',
-                           'lib/libvk_swiftshader.so',
-                           'lib/vk_swiftshader_icd.json',
-                           'swiftshader/libEGL.so',
-                           'swiftshader/libGLESv2.so'
                            ]
         if flavor == 'sdk':
             target['input'].append('nwjc')
             target['input'].append('payload')
             target['input'].append('chromedriver')
             target['input'].append('minidump_stackwalk')
-        if flavor in ['nacl','sdk'] :
-            target['input'] += ['nacl_helper', 'nacl_helper_bootstrap', 'pnacl']
-            if arch == 'x64':
-                target['input'].append('nacl_irt_x86_64.nexe')
-            elif arch == 'ia32':
-                target['input'].append('nacl_irt_x86_32.nexe')
-            else:
-                target['input'].append('nacl_irt_{}.nexe'.format(arch))
             
     elif platform_name == 'win':
         target['input'] = [
@@ -453,8 +441,6 @@ def make_packages(targets):
             os.mkdir(folder)
             if platform_name == 'linux':
                 os.mkdir(os.path.join(folder, 'lib'))
-            if platform_name in ['linux', 'win']:
-                os.mkdir(os.path.join(folder, 'swiftshader'))
             for f in t['input']:
                 src = os.path.join(binaries_location, f)
                 dest = os.path.join(folder, f)
