diff --git src/core/SkOpts.cpp src/core/SkOpts.cpp
index c09303e04a..bc62abf018 100644
--- src/core/SkOpts.cpp
+++ src/core/SkOpts.cpp
@@ -62,7 +62,7 @@ namespace SkOpts {
 
     #elif defined(SK_CPU_LOONGARCH)
         #if SK_CPU_LSX_LEVEL < SK_CPU_LSX_LEVEL_LASX
-            if (SkCpu::Supports(SkCpu::LOONGARCH_ASX)) { Init_lasx(); }
+            // if (SkCpu::Supports(SkCpu::LOONGARCH_ASX)) { Init_lasx(); }
         #endif
     #endif
         return true;
