diff --git gni/snapshot_toolchain.gni gni/snapshot_toolchain.gni
index e855b88e43..4abb839ec3 100644
--- gni/snapshot_toolchain.gni
+++ gni/snapshot_toolchain.gni
@@ -92,7 +92,7 @@ if (v8_snapshot_toolchain == "") {
       } else {
         _cpus = "x64_v8_${v8_current_cpu}"
       }
-    } else if (v8_current_cpu == "arm" || v8_current_cpu == "mipsel") {
+    } else if (v8_current_cpu == "arm" || v8_current_cpu == "mipsel" || v8_current_cpu == "loong64") {
       _cpus = "x86_v8_${v8_current_cpu}"
     } else {
       # This branch should not be reached; leave _cpus blank so the assert
